lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -456,7 +456,9 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
                 .variant_names
                 .insert(Symbol::Nonterminal(nt.clone()), name.clone());
         }
-
+        if self.grammar.intern_token.is_some() {
+            rust!(self.out, "{}phantom_data(::std::marker::PhantomData<&'input ()>),", self.prefix);
```

Comment:
If I understand correctly, this is adding a variant to the corresponding enum to hold phantom data. Would this be sufficient to  add this variant only when both `self.grammar.nonterminals` and `self.grammar.terminals` are empty, instead of adding unconditionally?